packages/ember-auto-import/ts/splitter.ts

```diff
@@ -315,6 +365,10 @@ export default class Splitter {
         ? this.options.bundles.bundleForPath(usage.path)
         : this.options.bundles.bundleForTreeType(usage.treeType);
 
+    if (usage.package.isLazyEngine) {
```

Comment:
I don't think this is correct. It only handles the case where the lazy engine itself uses ember-auto-import. But the lazy engine can use an addon that uses an addon that uses ember-auto-import. Should the lazy engine's bundle get those deps? Depends on who else consumes that chain of addons.

This is why I suggested making every package, not just lazy engines, emit their own corresponding module for webpack to process. It's much easier to reason about, because all our decision making is local-only, and the global reasoning can be left to webpack to optimize.